diff --git a/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp b/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
index b3446f1774..fe8b354603 100644
--- a/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
+++ b/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
@@ -301,7 +301,7 @@ EGLint EglGetErrorImpl(void)
 static __eglMustCastToProperFunctionPointerType FindBuiltinWrapper(const char* procname)
 {
 #if (defined(__aarch64__) || defined(__x86_64__))
-    static void* dlglv3Handle = dlopen("/system/lib64/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
+    static void* dlglv3Handle = dlopen("/system/lib64/platformsdk/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
 #else
     static void* dlglv3Handle = dlopen("/system/lib/platformsdk/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
 #endif
