diff --git a/services/distributeddataservice/service/object/BUILD.gn b/services/distributeddataservice/service/object/BUILD.gn
index e76a751..18db32c 100644
--- a/services/distributeddataservice/service/object/BUILD.gn
+++ b/services/distributeddataservice/service/object/BUILD.gn
@@ -78,6 +78,7 @@ ohos_source_set("distributeddata_object") {
     "hisysevent:libhisysevent",
     "kv_store:datamgr_common",
     "kv_store:distributeddb",
+    "hilog:libhilog",
   ]
 
   subsystem_name = "distributeddatamgr"
