diff --git a/audio/hdi_service/supportlibs/BUILD.gn b/audio/hdi_service/supportlibs/BUILD.gn
index 080713fca..5b897fa25 100644
--- a/audio/hdi_service/supportlibs/BUILD.gn
+++ b/audio/hdi_service/supportlibs/BUILD.gn
@@ -154,7 +154,7 @@ if (defined(ohos_lite)) {
         "$hdf_audio_path/supportlibs/alsa_adapter/src/alsa_lib_render.c",
         "$hdf_audio_path/supportlibs/alsa_adapter/src/alsa_snd_render.c",
         "$hdf_audio_path/supportlibs/alsa_adapter/src/alsa_soundcard.c",
-        "//device/board/${product_company}/${device_name}/audio_alsa/vendor_render.c",
+        # "//device/board/${product_company}/${device_name}/audio_alsa/vendor_render.c",
         "//third_party/cJSON/cJSON.c",
       ]
       if (defined(is_emulator) && is_emulator) {
