diff --git a/display/composer/v1_0/DisplayComposerType.idl b/display/composer/v1_0/DisplayComposerType.idl
index fce5368..0b77902 100644
--- a/display/composer/v1_0/DisplayComposerType.idl
+++ b/display/composer/v1_0/DisplayComposerType.idl
@@ -92,6 +92,7 @@ enum PixelFormat {
     PIXEL_FMT_BGRA_5551,                 /**< BGRA5551 format */
     PIXEL_FMT_BGRX_8888,                 /**< BGRX8888 format */
     PIXEL_FMT_BGRA_8888,                 /**< BGRA8888 format */
+    PIXEL_FMT_XRGB_8888,                 /**< XRGB8888 format */
     PIXEL_FMT_YUV_422_I,                 /**< YUV422 interleaved format */
     PIXEL_FMT_YCBCR_422_SP,              /**< YCBCR422 semi-planar format */
     PIXEL_FMT_YCRCB_422_SP,              /**< YCRCB422 semi-planar format */
@@ -305,6 +306,7 @@ enum InterfaceType {
     DISP_INTF_VGA,                  /**< VGA interface */
     DISP_INTF_MIPI,                 /**< MIPI interface */
     DISP_INTF_PANEL,                /**< PANEL interface */
+    DISP_INTF_VIRTUAL,              /**< VIRTUAL interface */
     DISP_INTF_BUTT,
 };
 
