diff --git a/build-profile.json5 b/build-profile.json5
index 7ada8fa..e96b540 100644
--- a/build-profile.json5
+++ b/build-profile.json5
@@ -18,7 +18,7 @@
       {
         "name": "default",
         "signingConfig": "default",
-        "compileSdkVersion": 14,
+        "compileSdkVersion": 15,
         "compatibleSdkVersion": 14,
         "targetSdkVersion": 14,
         "runtimeOS": "OpenHarmony",
