diff --git a/ohos_adapter/sensor_adapter/src/sensor_adapter_impl.cpp b/ohos_adapter/sensor_adapter/src/sensor_adapter_impl.cpp
index a6db240..d5e6d90 100644
--- a/ohos_adapter/sensor_adapter/src/sensor_adapter_impl.cpp
+++ b/ohos_adapter/sensor_adapter/src/sensor_adapter_impl.cpp
@@ -353,8 +353,8 @@ void SensorAdapterImpl::OhosSensorCallback(SensorEvent* event)
 
 int32_t SensorAdapterImpl::SubscribeOhosSensor(int32_t sensorTypeId, int64_t samplingInterval)
 {
-    WVLOG_I("SubscribeOhosSensor sensorTypeId: %{public}d, samplingInterval: %{public}lld",
-        sensorTypeId, samplingInterval);
+    // WVLOG_I("SubscribeOhosSensor sensorTypeId: %{public}d, samplingInterval: %{public}lld",
+    //     sensorTypeId, samplingInterval);
     WVLOG_I("SubscribeOhosSensor sensorTypeId: %{public}d", sensorTypeId);
     if (samplingInterval <= 0) {
         WVLOG_E("SubscribeOhosSensor error, samplingInterval is invalid.");
