diff --git a/mindspore-src/source/mindspore/ccsrc/plugin/device/cpu/kernel/nnacl/BUILD.gn b/mindspore-src/source/mindspore/ccsrc/plugin/device/cpu/kernel/nnacl/BUILD.gn
index 6b803e42..deeff094 100644
--- a/mindspore-src/source/mindspore/ccsrc/plugin/device/cpu/kernel/nnacl/BUILD.gn
+++ b/mindspore-src/source/mindspore/ccsrc/plugin/device/cpu/kernel/nnacl/BUILD.gn
@@ -58,7 +58,7 @@ config("nnacl_public_config") {
     ]
 
     # emulator not support avx512
-    if (!is_emulator) {
+    if (!is_emulator && product_name != "std_emulator") {
       cflags_c += [ "-mavx512f" ]
       defines += [ "ENABLE_AVX512" ]
     }
@@ -927,7 +927,7 @@ x86_64_only_sources = sse_avx_avx512_sources
 x86_64_only_sources += avx_fp32_kernel_sources
 
 # emulator not support avx512
-if (is_emulator) {
+if (is_emulator || product_name == "std_emulator") {
   not_needed(avx512_fp32_kernel_sources)
   not_needed(gemm_avx512_kernel_sources)
   not_needed(gemm_mask_avx512_kernel_sources)
diff --git a/mindspore-src/source/mindspore/lite/BUILD.gn b/mindspore-src/source/mindspore/lite/BUILD.gn
index cafc7f22..4ec06706 100644
--- a/mindspore-src/source/mindspore/lite/BUILD.gn
+++ b/mindspore-src/source/mindspore/lite/BUILD.gn
@@ -340,7 +340,7 @@ all_sources += missing_sources
 SUPPORT_NNRT = false
 
 # currently, only arm/arm64 real machine support nnrt
-if ((target_cpu == "arm" || target_cpu == "arm64") && !is_emulator) {
+if ((target_cpu == "arm" || target_cpu == "arm64") && !is_emulator && product_name != "std_emulator") {
   SUPPORT_NNRT = true
 }
 
@@ -430,7 +430,7 @@ ohos_shared_library("mindspore_lib") {
     ]
 
     # emulator not support avx512
-    if (!is_emulator) {
+    if (!is_emulator && product_name != "std_emulator") {
       defines += [ "ENABLE_AVX512" ]
     }
   }
diff --git a/mindspore-src/source/mindspore/lite/src/litert/kernel/cpu/BUILD.gn b/mindspore-src/source/mindspore/lite/src/litert/kernel/cpu/BUILD.gn
index 3ee357f6..8fb221d7 100644
--- a/mindspore-src/source/mindspore/lite/src/litert/kernel/cpu/BUILD.gn
+++ b/mindspore-src/source/mindspore/lite/src/litert/kernel/cpu/BUILD.gn
@@ -249,7 +249,7 @@ control_kernel_sources = [
 ]
 
 # emulator not support avx512
-if (!is_emulator) {
+if (!is_emulator && product_name != "std_emulator") {
   cpu_kernel_sources += [
     "fp32/convolution_im2col_avx512_fp32.cc",
     "fp32/matmul_fp32_avx512.cc",
@@ -342,7 +342,7 @@ ohos_source_set("cpu_kernel_obj") {
       "ENABLE_SSE",
       "ENABLE_AVX",
     ]
-    if (!is_emulator) {
+    if (!is_emulator && product_name != "std_emulator") {
       defines += [ "ENABLE_AVX512" ]
     }
   }
