diff --git a/static_core/BUILD.gn b/static_core/BUILD.gn
index b590507bf..71e5859a1 100644
--- a/static_core/BUILD.gn
+++ b/static_core/BUILD.gn
@@ -151,7 +151,7 @@ config("ark_config") {
   if (enable_codegen) {
     defines += [ "PANDA_COMPILER_ENABLE" ]
   }
-  if (is_emulator) {
+  if (is_emulator || product_name == "std_emulator") {
     defines += [ "PANDA_TARGET_EMULATOR" ]
   }
 
diff --git a/static_core/runtime/BUILD.gn b/static_core/runtime/BUILD.gn
index 81d006511..b32a72c1e 100644
--- a/static_core/runtime/BUILD.gn
+++ b/static_core/runtime/BUILD.gn
@@ -452,11 +452,6 @@ ohos_static_library("libarkruntime_static") {
 ohos_shared_library("libarkruntime") {
   deps = [ ":libarkruntime_set_static" ]
 
-  if (current_cpu == "amd64" || current_cpu == "x64" ||
-      current_cpu == "x86_64") {
-    ldflags = [ "-latomic" ]
-  }
-
   if (is_mingw || is_mac) {
     output_extension = "dll"
   } else {
