diff --git a/interfaces/innerkits/rust/netstack_rs/BUILD.gn b/interfaces/innerkits/rust/netstack_rs/BUILD.gn
index 74cc9999..237554a3 100644
--- a/interfaces/innerkits/rust/netstack_rs/BUILD.gn
+++ b/interfaces/innerkits/rust/netstack_rs/BUILD.gn
@@ -49,6 +49,7 @@ ohos_static_library("netstack_rs_cxx") {
   external_deps = [
     "rust_cxx:cxx_cppdeps",
     "cJSON:cjson",
+    "hilog:libhilog",
   ]
 
   part_name = "netstack"
