diff --git a/services/native/src/usb_host_manager.cpp b/services/native/src/usb_host_manager.cpp
index bd995fc9..08fceab0 100644
--- a/services/native/src/usb_host_manager.cpp
+++ b/services/native/src/usb_host_manager.cpp
@@ -1442,7 +1442,7 @@ static std::string Utf16leToUtf8(char *utf16leBytes, size_t length)
     }
     size_t charCount = length / HALF;
     // check and skip BOM
-    if (charCount > 0 && utf16leBytes[0] == 0xFF && utf16leBytes[1] == 0xFE) {
+    if (charCount > 0 && (unsigned char)utf16leBytes[0] == 0xFF && (unsigned char)utf16leBytes[1] == 0xFE) {
         charCount--;
         utf16leBytes += BOM_BYTE_COUNT;
         length -= BOM_BYTE_COUNT;
