diff --git a/frameworks/js/napi/common/BUILD.gn b/frameworks/js/napi/common/BUILD.gn
index 34985f08..94c5a213 100644
--- a/frameworks/js/napi/common/BUILD.gn
+++ b/frameworks/js/napi/common/BUILD.gn
@@ -54,7 +54,7 @@ ohos_static_library("inputmethod_js_common") {
   configs = [ ":inputmethod_js_common_config" ]
   public_configs = [ ":inputmethod_js_common_public_config" ]
   deps = [ "${inputmethod_path}/common:inputmethod_common" ]
-  external_deps = [ "napi:ace_napi" ]
+  external_deps = [ "napi:ace_napi", "hilog:libhilog" ]
   subsystem_name = "inputmethod"
   part_name = "imf"
 }
