diff --git a/storage_service_aafwk.gni b/storage_service_aafwk.gni
index 664413be..91f70770 100644
--- a/storage_service_aafwk.gni
+++ b/storage_service_aafwk.gni
@@ -41,7 +41,7 @@ declare_args() {
                                defined(global_parts_info.thirdparty_libusb) &&
                                defined(global_parts_info.thirdparty_libfuse)
 
-  enable_tee_client = !is_emulator
+  enable_tee_client = !is_emulator && product_name != "x86_general"
 }
 
 storage_manager_path =
