diff --git a/services/native/driver_extension/BUILD.gn b/services/native/driver_extension/BUILD.gn
index 8dbeb10..0e50ee5 100644
--- a/services/native/driver_extension/BUILD.gn
+++ b/services/native/driver_extension/BUILD.gn
@@ -75,6 +75,8 @@ ohos_shared_library("driver_extension_module") {
     "napi:ace_napi",
   ]
 
+  external_deps += [ "hilog:libhilog" ]
+
   relative_install_dir = "extensionability/"
   subsystem_name = "hdf"
   part_name = "external_device_manager"
