diff --git a/BUILD.gn b/BUILD.gn
index 6c471c7..e032da6 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -68,6 +68,17 @@ ohos_source_set("lzma_source_arm64") {
   sources += arm64_asm_source
 }
 
+# on device
+ohos_source_set("lzma_source_x86") {
+  configs = [ ":lzma_config_common" ]
+  public_configs = [ ":lzma_config_common" ]
+
+  include_dirs = [ "Asm/x86" ]
+  cflags = [ "-march=x86-64" ]
+
+  sources = common_c_source
+}
+
 # on host
 ohos_source_set("lzma_source_x86_host") {
   configs = [
@@ -126,7 +137,10 @@ ohos_shared_library("lzma_shared") {
     deps = [ ":lzma_source_arm64" ]
   } else if (target_cpu == "riscv64") {
     deps = [ ":lzma_source_riscv64" ]
+  } else if (target_cpu == "x86_64") {
+    deps = [ ":lzma_source_x86" ]
   }
+  
   innerapi_tags = [
     "chipsetsdk_sp_indirect",
     "platformsdk_indirect",
