diff --git a/audio/bundle.json b/audio/bundle.json
index e013a0b88..b95b69b6e 100755
--- a/audio/bundle.json
+++ b/audio/bundle.json
@@ -52,6 +52,7 @@
             "sub_component": [
                 "//drivers/peripheral/audio:hdi_audio",
                 "//drivers/peripheral/audio/effect:effect_model",
+                "//drivers/peripheral/audio/test/sample:idl_audio_sample",
                 "//drivers/peripheral/audio:libaudio_header_static"
             ],
             "inner_kits": [
diff --git a/audio/supportlibs/alsa_adapter/src/alsa_soundcard.c b/audio/supportlibs/alsa_adapter/src/alsa_soundcard.c
index e6bd6ebf1..570c036cf 100644
--- a/audio/supportlibs/alsa_adapter/src/alsa_soundcard.c
+++ b/audio/supportlibs/alsa_adapter/src/alsa_soundcard.c
@@ -1079,6 +1079,7 @@ int32_t SndElementWriteInt(
     snd_ctl_elem_id_t *elem_id = NULL;
     snd_ctl_elem_info_t *elem_info = NULL;
     snd_ctl_elem_value_t *elem_value = NULL;
+    unsigned int i, count;
 
     CHECK_NULL_PTR_RETURN_DEFAULT(cardIns);
     CHECK_NULL_PTR_RETURN_DEFAULT(ctlElem);
@@ -1105,10 +1106,15 @@ int32_t SndElementWriteInt(
 
     snd_ctl_elem_value_set_id(elem_value, elem_id);
     snd_ctl_elem_type_t type = snd_ctl_elem_info_get_type(elem_info);
+    count = snd_ctl_elem_info_get_count(elem_info);
     if (type == SND_CTL_ELEM_TYPE_INTEGER) {
-        snd_ctl_elem_value_set_integer(elem_value, 0, value);
+        for (i = 0; i < count; i++) {
+			snd_ctl_elem_value_set_integer(elem_value, i, value);
+		}
     } else if (type == SND_CTL_ELEM_TYPE_INTEGER64) {
-        snd_ctl_elem_value_set_integer64(elem_value, 0, (long long)value);
+        for (i = 0; i < count; i++) {
+			snd_ctl_elem_value_set_integer64(elem_value, i, (long long)value);
+		}
     } else {
         AUDIO_FUNC_LOGE("Element value is not integer type!\n");
         return HDF_FAILURE;
