diff --git a/services/native/src/power_mode_policy.cpp b/services/native/src/power_mode_policy.cpp
index 469871e..ebabb44 100644
--- a/services/native/src/power_mode_policy.cpp
+++ b/services/native/src/power_mode_policy.cpp
@@ -193,6 +193,11 @@ void PowerModePolicy::RemoveBackupMapSettingSwitch(uint32_t switchId)
 
 void PowerModePolicy::SavePowerModeRecoverMap()
 {
+    if (recoverMap_.empty()) {
+        POWER_HILOGW(FEATURE_POWER_MODE, "recoverMap_ is empty");
+        return;
+    }
+
     Json::Value recoverJson;
     for (const auto& pair : recoverMap_) {
         recoverJson[to_string(pair.first)] = pair.second;
