diff --git a/ohos_nweb/BUILD.gn b/ohos_nweb/BUILD.gn
index 6099676..3d943d5 100644
--- a/ohos_nweb/BUILD.gn
+++ b/ohos_nweb/BUILD.gn
@@ -23,7 +23,7 @@ if (target_cpu == "arm64") {
   web_source = "prebuilts/arm/ArkWebCore.hap"
   defines = []
 } else if (target_cpu == "x86_64") {
-  web_source = "prebuilts/arm64/ArkWebCore.hap"
+  web_source = "//vendor/oniro/common/hap/ArkWebCore-x86_64.hap"
   defines = [ "webview_x86_64" ]
 }
 
