diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index f857b0c5..50594d62 100755
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -365,8 +365,7 @@ is_wearable_product = "${target_platform}" == "wearable"
 
 is_intellitv_product = "${target_platform}" == "intellitv"
 
-if ((target_os == "ohos" && target_cpu == "x86_64") ||
-    device_company == "emulator") {
+if (device_company == "emulator") {
   is_emulator = true
 }
 
diff --git a/config/ohos/BUILD.gn b/config/ohos/BUILD.gn
index fceea5be..1b6f4e50 100755
--- a/config/ohos/BUILD.gn
+++ b/config/ohos/BUILD.gn
@@ -77,7 +77,7 @@ config("runtime_library") {
 
   # arm builds of libc++ starting in NDK r12 depend on unwind.
   if (current_cpu == "arm" || current_cpu == "arm64" ||
-      current_cpu == "riscv64") {
+      current_cpu == "riscv64" || current_cpu == "x86_64") {
     libs += [ "unwind" ]
   }
 
diff --git a/subsystem_config.json b/subsystem_config.json
index 8cbaee1e..b0693d49 100644
--- a/subsystem_config.json
+++ b/subsystem_config.json
@@ -222,5 +222,9 @@
   "llvmproject": {
     "path": "toolchain/llvm-project",
     "name": "llvmproject"
+  },
+  "oniro_soc_products": {
+    "path": "device/soc/oniro",
+    "name": "oniro_soc_products"
   }
 }
