diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 151f2edd..a038a959 100755
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -373,8 +373,7 @@ is_wearable_product = "${target_platform}" == "wearable"
 
 is_intellitv_product = "${target_platform}" == "intellitv"
 
-if ((target_os == "ohos" && target_cpu == "x86_64") ||
-    device_company == "emulator") {
+if (device_company == "emulator") {
   is_emulator = true
 }
 
diff --git a/config/ohos/BUILD.gn b/config/ohos/BUILD.gn
index 26b10a17..e59de7c2 100755
--- a/config/ohos/BUILD.gn
+++ b/config/ohos/BUILD.gn
@@ -77,7 +77,7 @@ config("runtime_library") {
 
   # arm builds of libc++ starting in NDK r12 depend on unwind.
   if (current_cpu == "arm" || current_cpu == "arm64" ||
-      current_cpu == "riscv64") {
+      current_cpu == "riscv64" || current_cpu == "x86_64") {
     libs += [ "unwind" ]
   }
 
diff --git a/ohos/images/build_image.py b/ohos/images/build_image.py
index d0ebca52..0ddb0f4d 100755
--- a/ohos/images/build_image.py
+++ b/ohos/images/build_image.py
@@ -41,6 +41,7 @@ def _prepare_root(system_path: str, target_cpu: str):
     _dir_list = [
         'config', 'dev', 'proc', 'sys', 'updater', 'system', 'vendor', 'data', 'chip_ckm',
         'storage', 'mnt', 'tmp', 'sys_prod', 'chip_prod', 'module_update', 'eng_system', 'eng_chipset'
+        'run', 'var', 'opt', 'usr'
     ]
     for _dir_name in _dir_list:
         os.makedirs(os.path.join(root_dir, _dir_name), exist_ok=True)
@@ -48,7 +49,7 @@ def _prepare_root(system_path: str, target_cpu: str):
     os.symlink('/system/bin/init', os.path.join(root_dir, 'init'))
     os.symlink('/system/etc', os.path.join(root_dir, 'etc'))
     os.symlink('/vendor', os.path.join(root_dir, 'chipset'))
-    if target_cpu == 'arm64' or target_cpu == 'riscv64':
+    if target_cpu == 'arm64' or target_cpu == 'riscv64' or target_cpu == 'x86_64':
         os.symlink('/system/lib64', os.path.join(root_dir, 'lib64'))
     os.symlink('/system/lib', os.path.join(root_dir, 'lib'))
 
diff --git a/subsystem_config.json b/subsystem_config.json
index 8cbaee1e..42ccfb8b 100644
--- a/subsystem_config.json
+++ b/subsystem_config.json
@@ -222,5 +222,9 @@
   "llvmproject": {
     "path": "toolchain/llvm-project",
     "name": "llvmproject"
+  },
+  "oniro_soc_products": {
+    "path": "device/soc/oniro",
+    "name": "oniro_soc_products"
   }
 }
