diff --git a/storage_service_aafwk.gni b/storage_service_aafwk.gni
index dbde16ab..fa56dc8f 100644
--- a/storage_service_aafwk.gni
+++ b/storage_service_aafwk.gni
@@ -39,8 +39,8 @@ declare_args() {
                                defined(global_parts_info.thirdparty_libusb) &&
                                defined(global_parts_info.thirdparty_libfuse)
 
-  enable_tee_client = !is_emulator
-  enable_huks_hdi = !is_emulator
+  enable_tee_client = !is_emulator && product_name != "x86_general"
+  enable_huks_hdi = !is_emulator && product_name != "x86_general"
 }
 
 storage_manager_path =
