diff --git a/storage_service_aafwk.gni b/storage_service_aafwk.gni
index 30b7f49e..a7ce28ad 100644
--- a/storage_service_aafwk.gni
+++ b/storage_service_aafwk.gni
@@ -41,8 +41,8 @@ declare_args() {
                                defined(global_parts_info.thirdparty_libfuse)
   storage_service_dfs_filemanager = defined(global_parts_info) &&
                                     defined(global_parts_info.filemanagement_dfs_service)
-  enable_tee_client = !is_emulator
-  enable_huks_hdi = !is_emulator
+  enable_tee_client = !is_emulator && product_name != "x86_general"
+  enable_huks_hdi = !is_emulator && product_name != "x86_general"
 }
 
 storage_manager_path =
