diff --git a/intention/boomerang/data/BUILD.gn b/intention/boomerang/data/BUILD.gn
index 0b2638521..970842455 100644
--- a/intention/boomerang/data/BUILD.gn
+++ b/intention/boomerang/data/BUILD.gn
@@ -9,7 +9,7 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
-# limitations under the License.
+# limitations under the License.
 
 import("../../../device_status.gni")
 
@@ -47,7 +47,7 @@ ohos_source_set("intention_boomerang_data") {
     "${device_status_root_path}/utils/common:devicestatus_util",
   ]
 
-  external_deps = [ "image_framework:image_native" ]
+  external_deps = [ "image_framework:image_native", "hilog:libhilog" ]
 
   subsystem_name = "${device_status_subsystem_name}"
   part_name = "${device_status_part_name}"
diff --git a/intention/drag/data/BUILD.gn b/intention/drag/data/BUILD.gn
index caa9c0f70..5af6b3543 100644
--- a/intention/drag/data/BUILD.gn
+++ b/intention/drag/data/BUILD.gn
@@ -52,6 +52,8 @@ ohos_source_set("intention_drag_data") {
     "image_framework:image_native",
   ]
 
+  external_deps += [ "hilog:libhilog" ]
+
   subsystem_name = "${device_status_subsystem_name}"
   part_name = "${device_status_part_name}"
 }
