From fc360daf28afbbc4389fa4ce5e7cea672f73fb2b Mon Sep 17 00:00:00 2001
From: "Pan, Kris" <kris.pan@intel.com>
Date: Fri, 29 Mar 2024 04:40:13 +0000
Subject: [PATCH] third_party/vk-gl-cts: add PTR_SIZE and CPU type for x86_64

Signed-off-by: Pan, Kris <kris.pan@intel.com>
---
 vk_gl_cts.gni | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/vk_gl_cts.gni b/vk_gl_cts.gni
index 30cf686a5..1fec8d225 100755
--- a/vk_gl_cts.gni
+++ b/vk_gl_cts.gni
@@ -61,7 +61,12 @@ if (target_cpu == "arm64") {
     "DE_PTR_SIZE=8",
     "DE_CPU=DE_CPU_ARM_64",
   ]
-} else {
+}  else if (target_cpu == "x86_64") {
+  deqp_common_defines += [
+    "DE_PTR_SIZE=8",
+    "DE_CPU=DE_CPU_X86_64",
+  ]
+}  else {
   deqp_common_defines += [
     "DE_PTR_SIZE=4",
     "DE_CPU=DE_CPU_ARM",
-- 
2.43.2

