diff --git a/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp b/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
index af40b70efd..7949f365a7 100644
--- a/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
+++ b/frameworks/opengl_wrapper/src/EGL/egl_wrapper_entry.cpp
@@ -300,7 +300,7 @@ EGLint EglGetErrorImpl(void)
 static __eglMustCastToProperFunctionPointerType FindBuiltinWrapper(const char* procname)
 {
 #if (defined(__aarch64__) || defined(__x86_64__))
-    static void* dlglv3Handle = dlopen("/system/lib64/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
+    static void* dlglv3Handle = dlopen("/system/lib64/platformsdk/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
 #else
     static void* dlglv3Handle = dlopen("/system/lib/platformsdk/libGLESv3.so", RTLD_NOW | RTLD_LOCAL);
 #endif
diff --git a/frameworks/opengl_wrapper/src/GLES3/gles3.cpp b/frameworks/opengl_wrapper/src/GLES3/gles3.cpp
index 53c1b9c249..2f239c5daf 100644
--- a/frameworks/opengl_wrapper/src/GLES3/gles3.cpp
+++ b/frameworks/opengl_wrapper/src/GLES3/gles3.cpp
@@ -26,7 +26,7 @@ template<typename Func = void*>
 Func GetEglApi(const char* procname)
 {
 #if (defined(__aarch64__) || defined(__x86_64__))
-    static const char* LIB_EGL = "/system/lib64/libEGL.so";
+    static const char* LIB_EGL = "/system/lib64/platformsdk/libEGL.so";
 #else
     static const char* LIB_EGL = "/system/lib/platformsdk/libEGL.so";
 #endif
diff --git a/rosen/modules/create_pixelmap_surface/BUILD.gn b/rosen/modules/create_pixelmap_surface/BUILD.gn
index ba4591f05f..b12b251590 100644
--- a/rosen/modules/create_pixelmap_surface/BUILD.gn
+++ b/rosen/modules/create_pixelmap_surface/BUILD.gn
@@ -48,6 +48,8 @@ ohos_source_set("create_pixelmap_surface_src") {
 
   external_deps = [ "graphic_surface:sync_fence" ]
 
+  external_deps += [ "hilog:libhilog" ]
+
   subsystem_name = "graphic"
   part_name = "graphic_2d"
 }
