diff --git a/arch/x86/mm/pat/memtype.c b/arch/x86/mm/pat/memtype.c
index 36b603d0c..b2f6abec0 100644
--- a/arch/x86/mm/pat/memtype.c
+++ b/arch/x86/mm/pat/memtype.c
@@ -986,22 +986,37 @@ static int get_pat_info(struct vm_area_struct *vma, resource_size_t *paddr,
  * If the vma has a linear pfn mapping for the entire range, we get the prot
  * from pte and reserve the entire vma range with single reserve_pfn_range call.
  */
-int track_pfn_copy(struct vm_area_struct *vma)
+int track_pfn_copy(struct vm_area_struct *dst_vma,
+		struct vm_area_struct *src_vma, unsigned long *pfn)
 {
 	resource_size_t paddr;
-	unsigned long vma_size = vma->vm_end - vma->vm_start;
+	unsigned long vma_size = src_vma->vm_end - src_vma->vm_start;
 	pgprot_t pgprot;
 
-	if (vma->vm_flags & VM_PAT) {
-		if (get_pat_info(vma, &paddr, &pgprot))
+	if (src_vma->vm_flags & VM_PAT) {
+		if (get_pat_info(src_vma, &paddr, &pgprot))
 			return -EINVAL;
 		/* reserve the whole chunk covered by vma. */
+		*pfn = paddr >> PAGE_SHIFT;
 		return reserve_pfn_range(paddr, vma_size, &pgprot, 1);
 	}
 
 	return 0;
 }
 
+void untrack_pfn_copy(struct vm_area_struct *dst_vma, unsigned long pfn)
+{
+	resource_size_t paddr;
+	unsigned long vma_size;
+
+	if (!(dst_vma->vm_flags & VM_PAT))
+		return;
+
+	paddr = (resource_size_t)pfn << PAGE_SHIFT;
+	vma_size = dst_vma->vm_end - dst_vma->vm_start;
+	free_pfn_range(paddr, vma_size);
+}
+
 /*
  * prot is passed in as a parameter for the new mapping. If the vma has
  * a linear pfn mapping for the entire range, or no vma is provided,
diff --git a/fs/binfmt_misc.c b/fs/binfmt_misc.c
index afb50f41a..a45b5ba12 100644
--- a/fs/binfmt_misc.c
+++ b/fs/binfmt_misc.c
@@ -816,7 +816,7 @@ static ssize_t bm_register_write(struct file *file, const char __user *buffer,
 
 	if (err) {
 		if (f) {
-			exe_file_allow_write_access(f);
+			allow_write_access(f);
 			filp_close(f, NULL);
 		}
 		kfree(e);
