diff --git a/frameworks/kits/js/common/picture_ndk/BUILD.gn b/frameworks/kits/js/common/picture_ndk/BUILD.gn
index 8115f37..94a2727 100644
--- a/frameworks/kits/js/common/picture_ndk/BUILD.gn
+++ b/frameworks/kits/js/common/picture_ndk/BUILD.gn
@@ -91,7 +91,7 @@ ohos_shared_library("picture") {
     "${image_subsystem}/interfaces/innerkits:image_native",
   ]
 
-  external_deps = [ "c_utils:utils" ]
+  external_deps = [ "c_utils:utils", "hilog:libhilog" ]
   if (enable_libexif) {
     external_deps += [ "libexif:libexif" ]
   }
